fix(core): fresh install fails — missing @lox-brain/team tsconfig reference

## Bug

Running the one-liner install script fails at build time:

```
curl -fsSL https://raw.githubusercontent.com/isorensen/lox-brain/main/scripts/install.sh | bash
```

### Error

```
src/mcp/index.ts:129:51 - error TS2307: Cannot find module '@lox-brain/team' or its corresponding type declarations.

129     const { registerTeamFeatures } = await import('@lox-brain/team');
                                                      ~~~~~~~~~~~~~~~~~

Found 1 error in src/mcp/index.ts:129
```

### Root cause

`packages/core/src/mcp/index.ts:129` dynamically imports `@lox-brain/team`, but:

1. `@lox-brain/team` is **not listed** in `packages/core/package.json` dependencies
2. `packages/core/tsconfig.json` has **no project reference** to `../team`
3. TypeScript resolves the import at compile time and fails because it can't find the type declarations

The import is inside a `try/catch` and gated by `LOX_MODE === 'team'`, so it works at runtime when the module isn't present — but `tsc` still type-checks it and rightfully errors.

### Expected behavior

`npm run build --workspaces` should succeed on a fresh clone so the installer can run.

### Fix approach

Either:
- **(A)** Add `@lox-brain/team` as an optional dependency + tsconfig reference in core (ensures types resolve)
- **(B)** Use a dynamic `import()` with `@ts-expect-error` or type-only import to suppress the check while keeping the runtime behavior

Option A is cleaner since the `team` package is already in the monorepo workspaces.

### Environment

- Fresh clone via `install.sh`
- Node.js 22+
- TypeScript 5.9.x
- npm workspaces

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+## [0.8.2] — 2026-04-07
+
+### Fixed
+- **Fresh install fails to build due to missing `@lox-brain/team` type declarations (#161).** Added ambient module declaration (`packages/core/src/types/lox-brain-team.d.ts`) so TypeScript can resolve the dynamic `import('@lox-brain/team')` in `mcp/index.ts` without adding a circular dependency between core and team packages. `npm run build --workspaces` now succeeds on a clean clone.
+
 ## [0.8.0] — 2026-04-06
 
 ### Added

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "lox-brain",
-  "version": "0.8.1",
+  "version": "0.8.2",
   "private": true,
   "description": "Lox — Where knowledge lives. Personal AI-powered Second Brain with semantic search, MCP Server, and Obsidian integration.",
   "workspaces": [

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@lox-brain/core",
-  "version": "0.8.1",
+  "version": "0.8.2",
   "private": true,
   "license": "MIT",
   "main": "dist/index.js",

--- a/packages/core/src/types/lox-brain-team.d.ts
+++ b/packages/core/src/types/lox-brain-team.d.ts
@@ -1,0 +1,43 @@
+/**
+ * Ambient module declaration for @lox-brain/team.
+ *
+ * Core uses a dynamic `await import('@lox-brain/team')` at runtime (team mode
+ * only) so it cannot have a direct package.json dependency — that would create
+ * a circular dependency since team already depends on core.  This declaration
+ * gives TypeScript enough type information to type-check the import without
+ * requiring the package to appear in core's dependency graph.
+ *
+ * Keep in sync with packages/team/src/index.ts.
+ */
+import type { LoxConfig } from '@lox-brain/shared';
+
+declare module '@lox-brain/team' {
+  export interface Tool {
+    name: string;
+    description: string;
+    inputSchema: Record<string, unknown>;
+    handler: (args: Record<string, unknown>) => Promise<unknown>;
+  }
+
+  export interface TeamRegistrationResult {
+    success: boolean;
+    org?: string;
+    peersRegistered?: number;
+    tools?: Tool[];
+    error?: string;
+  }
+
+  export function registerTeamFeatures(
+    server: unknown,
+    config: LoxConfig,
+    tools: Tool[],
+    publicKey: string,
+    options?: {
+      getClientIp?: () => string | null;
+      dbClient?: {
+        listRecent(options?: unknown): Promise<unknown>;
+        searchByAuthor(author: string, query?: string, options?: unknown): Promise<unknown>;
+      };
+    },
+  ): Promise<TeamRegistrationResult>;
+}

--- a/packages/installer/package.json
+++ b/packages/installer/package.json
@@ -1,6 +1,6 @@
 {
   "name": "lox",
-  "version": "0.7.2",
+  "version": "0.8.2",
   "private": true,
   "description": "Lox installer \u2014 set up your personal AI-powered Second Brain",
   "bin": {

--- a/packages/shared/package.json
+++ b/packages/shared/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@lox-brain/shared",
-  "version": "0.8.1",
+  "version": "0.8.2",
   "private": true,
   "license": "MIT",
   "main": "dist/index.js",

